system/security/LdapSecurity/ldaputils.cpp

```diff
@@ -79,8 +79,17 @@ LDAP* LdapUtils::LdapInit(const char* protocol, const char* host, int port, int
             DBGLOG("ldap_initialize error %s", ldap_err2string(rc));
             return nullptr;
         }
-        int reqcert = LDAP_OPT_X_TLS_NEVER;
-        ldap_set_option(NULL, LDAP_OPT_X_TLS_REQUIRE_CERT, &reqcert);
+
+        if (!isEmptyString(cipherSuite))
+        {
+            ldap_set_option(NULL, LDAP_OPT_X_TLS_CIPHER_SUITE, cipherSuite);
+        }
+        else
+        {
+            ldap_set_option(NULL, LDAP_OPT_X_TLS_CIPHER_SUITE, nullptr);
+            int reqcert = LDAP_OPT_X_TLS_NEVER;
+            ldap_set_option(NULL, LDAP_OPT_X_TLS_REQUIRE_CERT, &reqcert);
```

Comment:
Are we sure this isn't needed (for backward compatibility) when a cipher is supplied?  OMM it was needed to connect to the LDAP via a URL that didn't match the cert (IP address)?

Also since any `ldap_set_option(NULL, ...` is setting the default value to be used when the connection is established it might be safer to move it above the LDAP_INIT call - from some ldap docs:

_The ldap_init function returns the value for the ld parameter. If you use the ldap_set_option function before calling ldap_init and use NULL for the ld parameter, the values are set globally and copied to all LDAP session handles you create afterwards. If the ldap_set_option function is called after the ldap_init function, one of the following occurs:_

* _If the ld parameter is NULL, the values are set globally but do not affect the values in currently created LDAP session handles._
* _If the ld parameter is set to the value returned by the ldap_init function, the values are set for only that LDAP session handle._

Reply:
I agree, leave the existing call
ldap_set_option(NULL, LDAP_OPT_X_TLS_REQUIRE_CERT, &reqcert);
and add the if/else for the cipherSuite as a new section.
Should we also supply the protocol min/max ?

Reply:
I agree, I also seem to need the handle to be NULL in the cipherSuite call to get it to connect ok.